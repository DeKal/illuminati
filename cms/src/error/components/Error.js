import React from 'react'
import PropTypes from 'prop-types'

import Container from './base/ErrorContainer'
import Content from './base/ErrorContent'

import NotFound from 'error/components/NotFound'
import InternalServer from 'error/components/InternalServer'
import General from 'error/components/General'

const Error = ({ error }) => {
  return (
    <Container>
      <Content>{renderError(error)}</Content>
    </Container>
  )
}

const renderError = error => {
  const { code, message } = error
  switch (code) {
    case 404:
      return <NotFound message={message} />
    case 500:
      return <InternalServer message={message} />
    default:
    case undefined:
      console.error(error)
      return <General message={error.toString()} />
  }
}

Error.propTypes = {
  error: PropTypes.object
}

export default Error
