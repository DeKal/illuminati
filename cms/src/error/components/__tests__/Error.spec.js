import * as React from 'react'
import { shallow } from 'enzyme'
import Error from '../Error'

describe('Error', () => {
  it('should render 404 error page correctly', function() {
    const error = {
      code: 404,
      message: '404 error'
    }
    const wrap = shallow(<Error error={error} />)
    expect(wrap).toMatchSnapshot()
  })

  it('should render 500 error page correctly', function() {
    const error = {
      code: 500,
      message: '500 error'
    }
    const wrap = shallow(<Error error={error} />)
    expect(wrap).toMatchSnapshot()
  })

  it('should render general error page correctly', function() {
    const error = { text: 'some general error' }
    const wrap = shallow(<Error error={error} />)
    expect(wrap).toMatchSnapshot()
  })
})
