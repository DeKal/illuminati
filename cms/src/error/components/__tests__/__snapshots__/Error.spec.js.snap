// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should render 404 error page correctly 1`] = `
<Styled(div)>
  <Styled(div)>
    <NotFound
      message="404 error"
    />
  </Styled(div)>
</Styled(div)>
`;

exports[`Error should render 500 error page correctly 1`] = `
<Styled(div)>
  <Styled(div)>
    <InternalServer
      message="500 error"
    />
  </Styled(div)>
</Styled(div)>
`;

exports[`Error should render general error page correctly 1`] = `
<Styled(div)>
  <Styled(div)>
    <General
      message="[object Object]"
    />
  </Styled(div)>
</Styled(div)>
`;
